packages/typescript-estree/tests/lib/createProjectService.test.ts

```diff
@@ -1,22 +1,42 @@
+import fs from 'node:fs';
+import { join } from 'node:path';
+
 import * as ts from 'typescript';
 
 import { createProjectService } from '../../src/create-program/createProjectService';
+import { CORE_COMPILER_OPTIONS } from '../../src/create-program/shared';
 
-const mockReadConfigFile = jest.fn();
-const mockSetCompilerOptionsForInferredProjects = jest.fn();
+const FIXTURES_DIR = join(__dirname, '../fixtures/projectServicesComplex');
 
-jest.mock('typescript/lib/tsserverlibrary', () => ({
-  ...jest.requireActual('typescript/lib/tsserverlibrary'),
-  readConfigFile: mockReadConfigFile,
-  server: {
-    ProjectService: class {
-      setCompilerOptionsForInferredProjects =
-        mockSetCompilerOptionsForInferredProjects;
-    },
-  },
-}));
+const origSys = ts.sys;
+const origGetParsedCommandLineOfConfigFile =
+  ts.getParsedCommandLineOfConfigFile;
```

Comment:
[Testing] The rewritten test mocking here focuses in part  on mocking out `tsserver.getParsedCommandLineOfConfigFile`. But that function is now no longer being directly called by `createProjectService`. It's inside a `getParsedConfigFile`. I think the tests would be a lot more straightforward by:

* Moving any existing tests for what's now in `getParsedConfigFile` into a new `getParsedConfigFile.test.ts`.
  * This is one of my personal unit testing ideologies: when possible, testing a function's logic by calling the function, not one of the other functions that happens to call that function
* If there are any complex interactions between >2 functions that you'd _really_ like to test at a high level, making a dedicated integration test

It'd be a bit more work now, but IMO would result in better delineated unit tests that don't have to do quite so much deep mocking and un-mocking.